lib/_stream_readable.js

```diff
@@ -777,8 +777,13 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     debug('onerror', er);
     unpipe();
     dest.removeListener('error', onerror);
-    if (EE.listenerCount(dest, 'error') === 0)
-      errorOrDestroy(dest, er);
+    if (EE.listenerCount(dest, 'error') === 0) {
+      if (!dest.destroyed) {
+        errorOrDestroy(dest, er);
```

Comment:
@mcollina: this will not re-emit the error if the error comes from `destroy(err)`